macaroons/constraints.go

```diff
@@ -216,3 +216,25 @@ func HasCustomCaveat(mac *macaroon.Macaroon, customCaveatName string) bool {
 
 	return false
 }
+
+// GetCustomCaveatCondition returns the custom caveat condition for the given
+// custom caveat name from the given macaroon.
+func GetCustomCaveatCondition(mac *macaroon.Macaroon, customCaveatName string) (string, error) {
+	if mac == nil {
+		return "", fmt.Errorf("macaroon is nil")
+	}
+
+	caveatPrefix := []byte(fmt.Sprintf(
+		"%s %s", CondLndCustom, customCaveatName,
+	))
+	for _, caveat := range mac.Caveats() {
+		if bytes.HasPrefix(caveat.Id, caveatPrefix) {
+			if len(caveat.Id) > len(caveatPrefix) {
+				return string(caveat.Id[len(caveatPrefix)+1:]), nil
```

Comment:
If we do +1 then we also need to do that in the above condition. Otherwise a macaroon with just a space after the name would panic here.

Reply:
I thought this would be fine - if `len(caveat.Id)` is greater than `len(caveatPrefix)` then we should be able to reference `len(caveatPrefix) + 1` within `caveat.Id`. The line above is equivalent to `if len(caveat.Id) >= len(caveatPrefix) + 1` - in fact I'll just change it to that so it's a bit more clear. But I tested the code in its current state and the tests pass fine with a blank space for the caveat condition like `constraintFunc := macaroons.CustomConstraint("unit-test", " ")`.

Reply:
Could use a comment here outlining the expected format to aide a reader. 

How about something like this to make it more readable? 
```go
caveatSplit := strings.SplitN(caveat.Id, caveatPrefix, 2)
if len(caveatSplit) == 2 {
    return caveatSplit[1]
}
```

This'll split the string into two parts, one of everything before the prefix, and one of everything after. IIUC, we just want to return everything after.

Reply:
Assuming we don't want the whitespace, then we'd add a: `strings.TrimSpace(caveatSplit[1]))`

Reply:
I made this change and added a comment. Note that I needed to wrap `caveat.Id` and `caveatPrefix` with `string()` since they are currently `bytes`.

Rather than trim I added a character of whitespace to the prefix that we match/split on. That means we won't match a caveat with the same name but no condition (since it won't have that whitespace), but the end result is the same as we return an empty string in both cases.